### DISCUSSION TITLE
What direction should doom take?

### DISCUSSION THREAD

**Discussion by NTBBloodbath** [NTBBloodbath/doom-nvim#103] [Ideas]
Hey,  as will be known, doom seeks to have great extensibility so that it is not just a preconfigured setup, but also that you can configure it and adapt it to your needs, and that is very good.

However, I want to introduce a new design pattern that allows us to differentiate ourselves even more from the other setups; for this we currently have 2 options (and there may be more, feel comfortable to add another).

1. Be like Debian, an extremely stable environment but without the latest plugin features (we can also avoid dealing with breaking changes on them until the next doom-nvim releases)
2. Be like Gentoo, use only what you want, keeping the actual stability and the latest plugin features

---

For the option 1 we could pin each plugin to a specific commit hash in packer. In that way we can keep the plugins in a certain state / version, and pin them to a new version each doom-nvim release and fix the breaking changes if there are.

For the option 2 we will make doom even more modular, e.g. separate all the keybindings into different modules (e.g. one module for the windows management binds, other for the core, ect). And dynamically call them by using a Lua table in the configurations, like this 
```lua
-- doom_config.lua
...
  keybinds = {
    core = true,
    jump = true,
    plugins = true,
    windows = true,
  }
...
```
and so on with all the stuff that can be more modular, note that the provided example is just a prototype concept and is subject to changes.

Please let me know here what do you think about and what design should be better for us, thanks in advance.

Regards

**Comment by molleweide** [comment]
I feel like this is a bit above my level so my question might not make sense but wouldn't it work to combine the two options because both sound interesting

**Reply by bavalpey** [reply]
I agree with this.  The two options don't seem mutually exclusive.

**Reply by NTBBloodbath** [reply]
Hey, yeah that's what we're doing right now.

We recently merged #208 to pin plugins to a certain commits and have a more stable ecosystem internally. This behavior is also optional.

I was also making Doom more modular as you may know if you're using development branch and I'll try to make it even more modular because why not? lol

Feel free to suggest changes or additions :)

**Reply by molleweide** [reply]
Cool! I'm mjust trying to learn the core now as well as I can so that I understand what is actually going on. Also so that I can follow these kinds of discussions better.

I am looking fwd to the new mappings strategy/pattern.
